.github/workflows/build.yaml

```diff
@@ -0,0 +1,17 @@
+---
+on: [push, pull_request]
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+      - run: npm install
+      - run: npm run build
+      - run: npm run format
+      - uses: JamesIves/github-pages-deploy-action@releases/v4
+        if: github.event_name == 'push' && github.ref_name == 'main'
+        with:
+          branch: main
+          folder: .
```

Comment:
I would have removed this part as the deploy is managed thanks to tags as recommanded in https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management 

```suggestion
```

Reply:
Yes deploying to a separate branch makes more sense. I’d recommend naming that downstream branch as `release` rather than `gh-pages`, because the latter sounds like hosting the files over HTTP.